Add documentation about which flags are used when searching using the regex mode in the Sources tool (#2442)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/search/search-tool.md

<!-- ====================================================================== -->
## Search for regular expressions

> You can use regular expressions to find matching results.  To use a regular expression, click the **Use Regular Expression** (`.*`) button in the toolbar and enter a valid [JavaScript regular expressions](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Regular_Expressions) in the search input field.

Reviewer: ```suggestion
You can use regular expressions to find matching results.  To use a regular expression, click the **Use Regular Expression** (`.*`) button in the toolbar and enter a valid [JavaScript regular expression](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Regular_Expressions) in the search input field.
```

![Searching for a regular expression](../media/search-tool/search-tool-regexp.png)

> The slash characters that normally delimit regular expressions patterns in JavaScript aren't needed.

Reviewer: ```suggestion
The forward slash (`/`) characters that normally delimit regular expressions patterns in JavaScript aren't needed.
```

When using regular expressions, the **Search** tool matches results by using the `g`, `i`, and `m` regular expression flags:

* The `g` flag makes the search global, meaning that the tool will continue searching files even after a first match has been found.

> * The `i` flag makes the search case-insensitive. To disable this flag and only find results that match a particular case, click the **Match Case** (`Aa`) button in the search toolbar.

Reviewer: ```suggestion
* The `i` flag makes the search case-insensitive. To disable this flag and only find results that match a particular case, click the **Match Case** (**Aa**) button in the search toolbar.
```

> * The `m` flag specifies that the search should be multiline, meaning that the tool treats each line in the source files independently and the `^` and `$` symbols match the start and end of any line, respectively.

Reviewer: ```suggestion
* The `m` flag specifies that the search should be multiline, meaning that the tool treats each line in the source files independently.  When this flag is selected:
* The caret (`^`) symbol matches the start of any line.
* The dollar sign (`$`) symbol matches the end of any line.
```

> * The `g` flag makes the search global, meaning that the tool will continue searching files even after a first match has been found.
> * The `i` flag makes the search case-insensitive. To disable this flag and only find results that match a particular case, click the **Match Case** (`Aa`) button in the search toolbar.
> * The `m` flag specifies that the search should be multiline, meaning that the tool treats each line in the source files independently and the `^` and `$` symbols match the start and end of any line, respectively.

Reviewer: ```suggestion
| Flag | Name | Description |
|---|---|---|
| `g` | Global  | Makes the search global, meaning that the tool will continue searching files even after a first match has been found. |
| `i` | Case Insensitive | Ignores whether text is uppercase or lowercase. To disable this flag (even if it's included in the search string), and only find results that match a particular case, click the **Match Case** (**Aa**) button in the search toolbar. |
| `m` | Multiline | Specifies that the search should be multiline, meaning that the tool treats each line in the source files independently.  When this flag is selected, the caret (`^`) symbol matches the start of any line, and the dollar sign (`$`) symbol matches the end of any line. |
```

Author: Users can't input these flags in the search field. These are provided by default (only the `i` flag can be changed by using the match casing button `aA`).
Let me try to rephrase this entire part to make that clearer. I think this will address this comment, and the next.

Reviewer: you could use this table formatting if you want, and change wording.  i find table format faster to comprehend

> 

Reviewer: ```suggestion

See also:
* [Regular expressions](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Regular_Expressions) at MDN.

```

> 

Reviewer: Add examples of regex search strings showing g, i, or m.  This would be extremely helpful, to concretely see examples, instead of abstractly describing.  Any examples would make this much clearer.

> 

Reviewer: ```suggestion
Examples:

```javascript
regex using g; a string that you'd enter into Search text box
```

```javascript
regex using i; a string that you'd enter into Search text box
```

```javascript
regex using m; a string that you'd enter into Search text box
```

```
Add examples of regex search strings showing g, i, or m. This would be extremely helpful, to concretely see examples, instead of abstractly describing. Any examples would make this much clearer.

Regular expressions > Examples:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#examples

Author: I don't think we should add examples here. Users can't use these flags in DevTools. They are set automatically by the tool when you run a regular expression search (by clicking `.*`). The only one you can sort of change is the `i` flag by toggling the `aA` button. But otherwise, `m` and `g` are just always here.
The purpose of this section is to tell users that these flags are enabled for them, by default, so they can understand how their regular expressions get executed.

Reviewer: the change to "matches results as if the" addresses this comment

<!-- ====================================================================== -->
## Search for regular expressions

You can use regular expressions to find matching results.  To use a regular expression, click the **Use Regular Expression** (`.*`) button in the toolbar and enter a valid [JavaScript regular expression](https://developer.mozilla.org/docs/Web/JavaScript/Guide/Regular_Expressions) in the search input field.

![Searching for a regular expression](../media/search-tool/search-tool-regexp.png)

The forward slash (`/`) characters that normally delimit regular expressions patterns in JavaScript aren't needed.

Similarly, flags that optionally appear after the closing forward slash (`/`) in JavaScript regular expressions can't be used here. The **Search** tool matches results as if the `g`, `i`, and `m` regular expression flags had been provided:

* The search is global, meaning that the tool will continue searching files even after a first match has been found, as if the regular expression `g` flag had been provided.

> * The search ignores casing by default, as if the `i` flag had been provided. To disable this flag and only find results that match a particular case, click the **Match Case** (`Aa`) button in the search toolbar.

Reviewer: ```suggestion
* The search ignores casing by default, as if the `i` flag had been provided. To disable this flag and only find results that match a particular case, click the **Match Case** (**Aa**) button in the search toolbar.
```
UI control labels are shown in bold in this doc-set/repo.  Aa is bold in the UI I think, so bold would visually match the UI.  Backticks are for code in the context of a code listing.
